src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsGetSupportedGameFeaturesList Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+           UT_LOG_INFO("Result : Passed dsGetSupportedGameFeaturesList retrieved features_list count: %d",
+                 features_list.gameFeatureCount);
+    }
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to get AV latency.
+*
+* This test function is to get AV latency of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 015@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_avlatency(void)
+{
+    gTestID = 15;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int audio_latency = 0, video_latency = 0;
+   
+    ret = dsGetAVLatency(&audio_latency , &video_latency);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: dsGetAVLatency Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+           UT_LOG_INFO("Result : dsGetAVLatency audio_latency: %d ms, video_latency: %d ms",
+                         audio_latency , video_latency);
+    }
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to set allmsupport.
+*
+* This test function to set allmsupport of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 016@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_set_edid2allmsupport(void)
+{
+    gTestID = 16;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select;
+    bool allmsupport = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    printf(" \n================Please Select Inputs==================\n");
+
+    printf("\n*******Enter the port number to select*******\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select > numInputPorts)
+    {
+       printf("\nInvalid port selected\n");
+       goto exit;
+    }
+
+    port = select;
+
+    printf("\n*******Enter the allmsupport to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "true (i.e 1)\t"
+                     "false(i.e 0)\t"
+             "******************************************\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+    if(select < 0 || select > 1)
+    {
+       printf("\nInvalid allmsupport value selected\n");
+       goto exit;
+    }
+  
+    allmsupport = (bool)select;
+
+    ret = dsSetEdid2AllmSupport(port, allmsupport);
+    if(ret != dsERR_NONE)
```

Comment:
Remove the else case

In all cases, you need to see the results output

You've got some serious debugging issues todo if this was to assert and a failure of L1-L2 to fix in that case.

Reply:
done